### PR TITLE
Making generate() and generate_n conforming to C++20

### PR DESCRIPTION
**Pull request by hkaiser**

### CODE REVIEW

File: plugins/parcelport/mpi/parcelport_mpi.cpp [hunk 1]
```diff
@@ -284,7 +284,7 @@ namespace hpx { namespace traits
 #else
                 "env = ${HPX_HAVE_PARCELPORT_MPI_ENV:"
                         "MV2_COMM_WORLD_RANK,PMIX_RANK,PMI_RANK,OMPI_COMM_WORLD_SIZE,"
-                        "ALPS_APP_PE"
+                        "ALPS_APP_PE,PMIX_RANK"
```
Comment (msimberg) [inline plugins/parcelport/mpi/parcelport_mpi.cpp:287]: Did this one sneak in by mistake? Looks like `PMIX_RANK` is already in the list anyway...

File: libs/algorithms/tests/unit/container_algorithms/generate_range.cpp [hunk 2]
```diff
@@ -52,7 +77,7 @@ void test_generate(ExPolicy policy, IteratorTag)
 }
 
 template <typename ExPolicy, typename IteratorTag>
-void test_generate_async(ExPolicy p, IteratorTag)
+void test_generate_async(ExPolicy&& p, IteratorTag)
```
Comment (gonidelis) [inline libs/algorithms/tests/unit/container_algorithms/generate_range.cpp:80]: @hkaiser  Could you please explain why the `&&`?

Comment (hkaiser) [reply libs/algorithms/tests/unit/container_algorithms/generate_range.cpp:80]: In this case it doesn't really make a big difference, I just thought that using a forwarding reference here would be more idiomatic and avoids copying the execution policy.

File: libs/algorithms/tests/unit/algorithms/generate.cpp [hunk 3]
```diff
@@ -187,7 +245,7 @@ void generate_exception_test()
 
 //////////////////////////////////////////////////////////////////////////////
 template <typename ExPolicy, typename IteratorTag>
-void test_generate_bad_alloc(ExPolicy policy, IteratorTag)
+void test_generate_bad_alloc(ExPolicy&& policy, IteratorTag)
```
Comment (gonidelis) [inline libs/algorithms/tests/unit/algorithms/generate.cpp:248]: @hkaiser  Why no tests for bad_alloc without exec-policy?

Comment (hkaiser) [reply libs/algorithms/tests/unit/algorithms/generate.cpp:248]: I was too lazy to add those ;-) (I have not added those for any of the ported algorithms, sorry.

**Pull request merged**